tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -1386,27 +1429,38 @@ SQLITE_API int sqlite3_result_zeroblob64(sqlite3_context *, sqlite3_uint64 n) {
 }
 
 // TODO complain
-const void *sqlite3_value_blob(sqlite3_value *) {
-	return nullptr;
+const void *sqlite3_value_blob(sqlite3_value *pVal) {
+	return sqlite3_value_text(pVal);
 }
 
-double sqlite3_value_double(sqlite3_value *) {
-	return 0;
+double sqlite3_value_double(sqlite3_value *pVal) {
+	if (pVal->type == SQLiteTypeValue::FLOAT) {
```

Comment:
I believe we should try to cast the value in the case the underlying value is an integer or text value, see e.g. [the docs](https://sqlite.org/c3ref/value_blob.html):

> As long as the input parameter is correct, these routines can only fail if an out-of-memory error occurs during a format conversion. Only the following subset of interfaces are subject to out-of-memory errors:
sqlite3_value_blob()
sqlite3_value_text()
sqlite3_value_text16()
sqlite3_value_text16le()
sqlite3_value_text16be()
sqlite3_value_bytes()
sqlite3_value_bytes16()